addons/web/static/src/core/py_js/py_date.js

```diff
@@ -36,15 +36,17 @@ function divmod(a, b, fn) {
     return fn(Math.floor(a / b), mod);
 }
 
-function assert(bool) {
+function assert(bool, message = "AssertionError") {
     if (!bool) {
-        throw new AssertionError("AssertionError");
+        throw new AssertionError(message);
     }
 }
 
 const DAYS_IN_MONTH = [null, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
 const DAYS_BEFORE_MONTH = [null];
 
+const DAYS_IN_YEAR = [31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334, 366];
```

Comment:
is it based on DAYS_IN_MONTH? If yes, don't you need to put 365 on last element (year based on 28 days in february)

Reply:
no it is not, this will serve to determine the real day of year afterwards. It is the way the alo uses it that gives its sense to tis list.
Besides, the algo and list are identitcal to the implementation in python